pkgs/development/compilers/lunarml/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, fetchFromGitHub
+, fetchzip
+, stdenvNoCC
+  # build inputs
+, mlton
+  # check inputs
+, lua5_3
+, luajit
+, nodejs
+  # flags
+, withExamples ? false
+}:
+
+let
+  name = "lunarml";
+in
+stdenvNoCC.mkDerivation {
+  inherit name;
+
+  version = "unstable-2022-01-24";
+
+  src = fetchFromGitHub {
+    owner = "minoki";
+    repo = "LunarML";
+    rev = "60b646ae7d55550626642a1b3f9a0af3944cc2a6";
+    sha256 = "sha256-lI947OQlb5s9tkSns9bP4uNyjsa1fsPuPiczgfVRe1I=";
+  };
+
+  nativeBuildInputs = [
+    mlton
+  ];
+
+  checkInputs = [
```

Comment:
You can now use [`nativeCheckInputs`](https://github.com/NixOS/nixpkgs/pull/206742)